kedro/pipeline/modular_pipeline.py

```diff
@@ -108,8 +114,17 @@ def pipeline(
             any of the expected types (str, dict, list, or None).
 
     Returns:
-        A new ``Pipeline`` object with the new nodes, modified as requested.
+        A new ``Pipeline`` object.
     """
+    if isinstance(pipe, Pipeline):
+        # To ensure that we are always dealing with a *copy* of pipe.
+        pipe = Pipeline([pipe], tags=tags)
```

Comment:
`Pipeline` only takes an iterable of pipelines/nodes, hence wrapping this in `[]`.